tests/cases/conformance/types/tuple/named/partiallyNamedTuples.ts

```diff
@@ -0,0 +1,44 @@
+type NamedAndAnonymous = [a: string, number];
+
+function fa1(...args: NamedAndAnonymous) {}
+function fa2(a: NamedAndAnonymous, ...args: NamedAndAnonymous) {}
+
+type NamedAnonymousMixed = [a: string, number, c: number, NamedAndAnonymous];
+
+function fb1(...args: NamedAnonymousMixed) {}
+function fb2(a: NamedAnonymousMixed, ...args: NamedAnonymousMixed) {}
+function fb3(a: NamedAnonymousMixed, ...args: NamedAnonymousMixed[3]) {}
+
+type ToAnonymousTuple<T extends unknown[]> = {
+  [K in keyof T]: [K, T[K], keyof T, T];
+};
+
+type AnonymousToAnonymous = ToAnonymousTuple<[boolean, number]>;
+type MixedToAnonymous = ToAnonymousTuple<[boolean, second: number]>;
```

Comment:
When I proposed extra tests for mapped types I meant ones like this `MixedToAnonymous` or `MixedToMixed`. I feel like the rest is somewhat redundant cause it doesn't exactly test the new behavior - it doesn't matter what is in the "template" of the mapped type since that's mostly unrelated to the mapped type itself.

Reply:
To rephrase - the only behavior that I consider new for mapped types in relation to his PR is that here we are able to preserve the labels:
```ts
type Mapped<T extends any[]> = {
  [K in keyof T]: 1
}
type Res = Mapped<[first: true, number]>
```

Given that a mapped type can't add or remove tuple elements, the surface area of the change for mapped types is pretty low.